components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -457,6 +460,37 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return user;
     }
 
+    public async needsVerification(ctx: TraceContext): Promise<boolean> {
+        const user = this.checkUser("needsVerification");
+        return this.phoneVerificationService.needsVerification(user);
+    }
+
+    public async sendPhoneNumberVerificationToken(ctx: TraceContext, phoneNumber: string): Promise<void> {
+        this.checkUser("sendPhoneNumberVerificationToken");
+        return this.phoneVerificationService.sendVerificationToken(phoneNumber);
+    }
+
+    public async verifyPhoneNumberVerificationToken(
+        ctx: TraceContext,
+        phoneNumber: string,
+        token: string,
+    ): Promise<boolean> {
+        const user = this.checkUser("verifyPhoneNumberVerificationToken");
+        const checked = await this.phoneVerificationService.verifyVerificationToken(phoneNumber, token);
+        if (!checked) {
+            return false;
+        }
+        if (!user.additionalData) {
+            user.additionalData = {};
+        }
+        if (!user.additionalData?.profile) {
```

Comment:
nit: this cascade could be wrapped in a `User.updateProfile(partial: Partial<User.Profile>)` method.